lib/engine/auto_router.rb

```diff
@@ -47,15 +47,33 @@ def compute(corporation, **opts)
           chain = []
           left = nil
           right = nil
+          last_left = nil
+          last_right = nil
 
           complete = lambda do
             chains << { nodes: [left, right], paths: chain }
+            last_left = left
+            last_right = right
             left, right = nil
             chain = []
           end
 
-          assign = lambda do |n|
-            if !left
+          assign_both = lambda do |a, b|
```

Comment:
can we delete this lambda?

Reply:
OK. It will make the other larger, but sure.